src/models/db/subscriptions.js

```diff
@@ -9,12 +9,11 @@ function * getSubscription (id, options) {
   return (yield Subscription.findById(id, options))
 }
 
-function * getMatchingSubscription (subscription, options) {
-  return (yield Subscription.findOne({where: {
+function * getMatchingSubscriptions (subscription, options) {
+  return (yield Subscription.findWhere({
```

Comment:
Yes, the name is because it returns an array type of length 0 or 1. I'll refactor to make it more clear.